src/pages/[id]/index.js

```diff
@@ -19,23 +19,35 @@ import {
   HEIGHT_40PX,
 } from '@constants/profile-image';
 import { useEffect, useState } from 'react';
+import { userContext } from '@store/user/user-context';
 
 const MemberProfile = ({ imageLink, user, contributions, errorMessage }) => {
+  const { isSuperUserMode } = userContext();
   const [activeTasksData, setActiveTasksData] = useState([]);
   const router = useRouter();
   const { id } = router.query;
-  useEffect(() => {
-    (async () => {
-      const tasksURL = getActiveTasksURL(id);
-      const tasksResponse = await fetch(tasksURL);
-      const { tasks } = await tasksResponse.data;
-      setActiveTasksData(tasks);
-    })();
-  }, [id]);
 
   if (errorMessage) {
     return <NotFound errorMsg={errorMessage} />;
   }
+  if (!user.roles?.member && !isSuperUserMode) {
+    return <NotFound errorMsg="Unauthorized to view this page." />;
```

Comment:
Is this better to save this as constants if we are planning to utilize it in different pages?

Reply:
yes, it's better to add this as a constant. I have created a new constant for the same now :).